z3/src/z3_factory.cpp

```diff
@@ -1,15 +1,48 @@
+/*********************                                                        */
+/*! \file z3_factory.cpp
+** \verbatim
+** Top contributors (to current version):
+**   Amalee Wilson
+** This file is part of the smt-switch project.
+** Copyright (c) 2020 by the authors listed in the file AUTHORS
+** in the top-level source directory) and their institutional affiliations.
+** All rights reserved.  See the file LICENSE in the top-level source
+** directory for licensing information.\endverbatim
+**
+** \brief Factory for creating a Z3 SmtSolver
+**
+**
+**/
+
 #include "z3_factory.h"
 
 #include "z3_solver.h"
 
-namespace smt
-{
+#include "logging_solver.h"
+
+namespace smt {
 
-/* Z3SolverFactory implementation */
-SmtSolver Z3SolverFactory::create()
+bool initialized = false;
+
+/* Z3SolverFactory implementation with logging */
+SmtSolver Z3SolverFactory::create(bool logging)
 {
-  return std::make_unique<Z3Solver>();
+  // Must initialize only once.
+  // Different instances of the solver will have
+  // different contexts.
+  // if (!initialized)
+  // {
+  //   yices_init();
+  //   initialized = true;
+  // }
+
+  SmtSolver solver = std::make_shared<Z3Solver>();
+  // if (logging)
```

Comment:
You should be able to put this commented code in without any concerns. Although I could be missing something so if it causes problems you can leave it out until the implementation is done.

Reply:
I took it out initially because I was having problems in general with creating a solver and wanted to simplify the file. I forgot to put it back in, thanks